### PR TITLE
UX: Icons: Remove legacy connect icon

### PR DESCRIPTION
**Pull request by darkwing**
## Explanation

Removes and replaces the old Connect icon in the accounts dropdown.

## Screenshots/Screencaps

new-connect

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by CharlesBlaqk** [comment]
We'd be happy to look into this for you. kindly file a support ticket below 
https://forms.gle/Wmpm6NLzYm9uXYuU9

File: ui/components/app/account-menu/account-menu.component.js [hunk 1]
```diff
@@ -412,10 +411,7 @@ export default class AccountMenu extends Component {
             }
           }}
           icon={
-            <IconConnect
-              color="var(--color-icon-alternative)"
-              ariaLabel={t('connectHardwareWallet')}
-            />
+            <Icon name={ICON_NAMES.HARDWARE} color={Color.iconAlternative} />
```
Comment (NidhiKJha) [inline ui/components/app/account-menu/account-menu.component.js:414]: I think we should keep the `ariaLabel`

**Review by NidhiKJha — COMMENTED** [review]
Looking great, Just a small nit

File: ui/components/app/account-menu/account-menu.component.js (hunk 1 above)
Comment (darkwing) [reply ui/components/app/account-menu/account-menu.component.js:414]: `ariaLabel` isn't a prop of `<Icon>`

**Review by NidhiKJha — APPROVED** [review]

File: ui/components/app/account-menu/account-menu.component.js (hunk 1 above)
Comment (NidhiKJha) [reply ui/components/app/account-menu/account-menu.component.js:414]: Duh, My bad 🙈

**Review by pedronfigueiredo — APPROVED** [review]

**Pull request merged**